kv/kvserver: TestStoreRangeLease failed

kv/kvserver.TestStoreRangeLease [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20890963?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20890963?buildTab=artifacts#/) on release-26.1 @ [e337c1076f14556c7a8ef47929e3ce9d30cb3e00](https://github.com/cockroachdb/cockroach/commits/e337c1076f14556c7a8ef47929e3ce9d30cb3e00):


```
=== RUN   TestStoreRangeLease
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/e6a7b3ae433fb38f3b322ee831772836/logTestStoreRangeLease2597600017
    test_log_scope.go:82: use -show-logs to present logs inline
    client_lease_test.go:121: desc: r4:/System/{NodeLivenessMax-tsd} [(n1,s1):1, next=2, gen=0, sticky=9223372036.854775807,2147483647]
    client_lease_test.go:121: desc: r81:{a-b} [(n1,s1):1, next=2, gen=2, sticky=9223372036.854775807,2147483647]
    client_lease_test.go:121: desc: r82:{b-c} [(n1,s1):1, next=2, gen=3, sticky=9223372036.854775807,2147483647]
    client_lease_test.go:121: desc: r83:{c-/Table/0} [(n1,s1):1, next=2, gen=3, sticky=9223372036.854775807,2147483647]
    client_lease_test.go:124: expected lease type LeaseLeader; got repl=(n1,s1):1 seq=1 start=0,0 type=LeaseExpiration exp=1766221308.352665601,0 pro=1766221302.352665601,0 acq=Request for key "c"
    client_lease_test.go:127: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e6a7b3ae433fb38f3b322ee831772836/logTestStoreRangeLease2597600017
--- FAIL: TestStoreRangeLease (1.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeLease.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58086

--- a/pkg/kv/kvserver/client_lease_test.go
+++ b/pkg/kv/kvserver/client_lease_test.go
@@ -93,6 +93,21 @@ func TestStoreRangeLease(t *testing.T) {
 		tc.SplitRangeOrFatal(t, splitKey)
 	}
 
+	// Wait until everyone has a raft leader. We've seen test failures that result
+	// from the RHS of the final split not yet having a leader when we request a
+	// lease and getting an expiration based lease.
+	for _, key := range splitKeys {
+		repl := store.LookupReplica(roachpb.RKey(key))
+		testutils.SucceedsSoon(t, func() error {
+			status := repl.RaftStatus()
+			if status == nil || status.Lead == 0 {
+				return errors.Errorf("waiting for raft leadership on key %s (range %d): state=%v",
+					key, repl.RangeID, status.RaftState)
+			}
+			return nil
+		})
+	}
+
 	// Expire all leases and send a write request to trigger a lease acquisitions.
 	// At this point, we have a leader, so the lease acquisition should be for a
 	// leader lease.

